project/index.mjs

```diff
@@ -138,6 +138,94 @@ export function respondWithProject(req, res, project) {
   res.location(id).status(200).send(retVal)
 }
 
+/**
+ * Check for valid project keys and create defaults, then send to database
+ * @param {Object} req The request object
+ * @param {Object} res The response object
+ */
+async function createNewProject(req, res) {
+  let project = req.body
+  /* 
+  TODO:
+    - Add creator based on authenticated user
+    - Add group
+    - Add slug (and figure out a good way to format it)
+    - Check for manifest, create one in the database if it doesn't exist
+  */
+
+  // TEMPORARY stubbing out required keys we need to generate
+  project.creator = "user"
+  project.group = "group"
+  project.slug = "slug"
+
+  // Required keys
+  if (project.created) {
```

Comment:
Yes something like the Project class could have a function to validate the minimum requirements.

Reply:
Should I put this functionality into a Project class then instead of in the project/index.mjs file where it is now?

Reply:
We will want a ProjectFactory that will prepare a database record. This will give us the ability to throw an invalid error with details, but also to assert lots of defaults. For example, `req.user.agent` will be the `creator`, `created: now` can be automatic, if there is no `title` a placeholder can be applied, a Group and OWNER can be applied, etc. It's way more complex than just "is this the minimum?" so let's leave it boxed up for the moment.